### DISCUSSION TITLE
Automatic sort order in my collection

### DISCUSSION THREAD

**Discussion by bios4** [devemux86/cruiser#551] [Q&A] [answered]
Currently running the latest Cruiser 4.1.0-b2...

After importing a new entry in "My Collection" (tracks in that particular example), the new entry is immediately sorted right after the existing entry where the "three-dots-menu" was used to trigger the import. See screenshots:

<img src="https://github.com/devemux86/cruiser/assets/145460147/ee2c51da-df2b-4bf6-973c-e3cff436551d" width="33%">
<img src="https://github.com/devemux86/cruiser/assets/145460147/b7e5fc29-5b58-4346-9b14-661a3303f2fc" width="33%">

ALTOUGH my sort order is set differently ("sort by name" in my case), the new item is NOT properly sorted in the existing list after import. Additionally, the new item STAYS at that very position even after reloading "My collection" and/or a complete restart of the app.

My wish would be:
- Every import operation takes the set sort order into account
- After the import, the new element is inserted and displayed in the “correct” position (regardless of whether the sort order is set to “name”, “length” or “duration”); without the need to apply the sort order manually once more
- The selected sort order is taken into account each time a folder in “My collection” is reloaded (which is currently not the case obviously because my new inserted item remains in the position where it was added after import)

Question to @devemux86: could this behaviour be changed/implemented (applicable to all item types in "My collection")?

**Comment by devemux86** [comment]
@bios4 thanks for the suggestion.

>  my sort order is set differently ("sort by name" in my case)

This is not a saved setting to automatically sort the items.

Sort currently is an action, the sort happens when we call it.
So we can reorder the items and they stay at their positions.

What you suggest is to have an automatic sort option that automatically sorts the items?
It this case the reorder via drag up / down should be deactivated.

**Reply by bios4** [reply]
Thanks for the quick reply Emux! 😄

> What you suggest is to have an automatic sort option that automatically sorts the items?

Sort of, yes - but...

> It this case the reorder via drag up / down should be deactivated.

...not neccessarily!

What I thought of is:

- When a list in "My collection" is loaded for the first time the entries are sorted "as is"
- If a user selects a sort order via three-dots-menu, this sort order is applied to (and persistently stored for) that particular list
  - If an entry is then added/imported, it should be added/sorted at the proper position
- If a user (re)arranges (single) entries per drag & drop, this "custom" sort order is applied (and persistently saved) until a new sort order is selected (either via three-dots-menu or per drag & drop)
  - If an entry is added/imported to a "custom sorted" list, it could be added at the very bottom of the list until the user sorts it in the way he wants it to be

But unfortunately I have absolutely NO idea if this is even possible.. ❓

**Reply by devemux86** [reply]
Quick note about only the import:

> If an entry is added/imported to a "custom sorted" list, it could be added at the very bottom of the list until the user sorts it in the way he wants it to be

Currently the imported items are placed after the location of the menu that initiated the import,
so we can import items at selected locations.

We can change it and always import items at the end of the list, but that will remove the above flexibility.
As we can already import items at the end of the list if use the menu of the last item.

**Reply by bios4** [reply]
> Currently the imported items are placed after the location of the menu that initiated the import

I know, that's what I wrote in my initial post...

But in sorted lists it could be at least really hard (or even almost impossible, e.g. when sorted "by distance of current location" in Favorites) to find the "correct" list item which is required to select the three dots for the import to have the new item in the "correct" place.

Therefore I asked for possibilities to change this. 😉

> As we can already import items at the end of the list if use the menu of the last item.

True, this could be at least an existing workaround (even when having to scroll to the bottom of a long list first) for custom sorted lists, but does not help with non-custom-sorted lists. Because then one has to re-sort again manually.

**Comment by bios4** [comment]
I just learned that the behaviour when importing items to "My collection" was changed in the meantime...

An import is now NOT anymore triggered by clicking the "three dots menu" of an entry in an existing list, but by clicking the "three dots menu" of a (parent) group. The imported item is then sorted at the very end of the list in that group.

This is a nice solution, thanks for implementing!

**Comment by devemux86** [comment]
Automatic sort order in my collection will be available in the next Cruiser.

It works as described [above](https://github.com/devemux86/cruiser/discussions/551#discussioncomment-9557425).

Favorites|Routes / Tracks
-|-
![Favorites](https://github.com/user-attachments/assets/bdfe0ca1-228e-472c-aaa6-4f888c59f80f)|![Routes](https://github.com/user-attachments/assets/2b380ffa-93a0-4060-bf36-770df300395a)

**Comment by devemux86** [comment, marked as answer]
- Implemented in https://github.com/devemux86/cruiser/discussions/901

It works as described [above](https://github.com/devemux86/cruiser/discussions/551#discussioncomment-9557425).
